### PR TITLE
OSSM-3734: Move header-append-filter push image postsubmit jobs

### PR DESCRIPTION
**Pull request by yxun**

### CODE REVIEW

File: ci-operator/config/maistra/header-append-filter/maistra-header-append-filter-maistra-2.2.yaml [hunk 1]
```diff
@@ -20,6 +20,102 @@ tests:
...
+  cluster_claim:
+    architecture: amd64
+    cloud: aws
+    owner: openshift-ci
+    product: ocp
+    timeout: 1h0m0s
+    version: "4.10"
+  steps:
+    env:
+      MAISTRA_BUILDER_IMAGE: registry.ci.openshift.org/ci/maistra-builder:2.4
```
Comment (jwendell) [inline ci-operator/config/maistra/header-append-filter/maistra-header-append-filter-maistra-2.2.yaml:33]: It looks like 2.4 is used everywhere. Is it on purpose?

**Review by jwendell — COMMENTED** [review]
LGTM, feel free to add the postsubmit:true back

File: ci-operator/config/maistra/header-append-filter/maistra-header-append-filter-maistra-2.2.yaml (hunk 1 above)
Comment (yxun) [reply ci-operator/config/maistra/header-append-filter/maistra-header-append-filter-maistra-2.2.yaml:33]: that's from maistra/test-infra. I was wondering why as well. When I saw the builder image tag 2.4 in test-infra, I try to keep it as it was.

**Review by jwendell — APPROVED** [review]

**Pull request merged**